client/src/pages/searches/MySavedSearches.tsx

```diff
@@ -58,77 +66,110 @@ const MySavedSearches = ({
     pageDispatchers
   })
   usePageTitle("My Saved Searches")
+
+  const savedSearches = useMemo(() => data?.savedSearches || [], [data])
+  const totalCount = savedSearches.length
+  const paginatedSearches = useMemo(
+    () =>
+      savedSearches.slice(
+        pageNum * DEFAULT_PAGESIZE,
+        (pageNum + 1) * DEFAULT_PAGESIZE
+      ),
+    [savedSearches, pageNum]
+  )
+
+  useEffect(() => {
+    const newQueries = Object.assign({})
+    paginatedSearches.forEach(search => {
+      const objType = SEARCH_OBJECT_TYPES[search.objectType]
+      const queryParams = utils.parseJsonSafe(search.query)
+      deserializeQueryParams(
+        objType,
+        queryParams,
+        (objectType, filters, text) => {
+          newQueries[search.uuid] = { objectType, filters, text }
+          setDeserializedQueries(prev => ({ ...prev, ...newQueries }))
+        }
+      )
+    })
+  }, [paginatedSearches])
+
   if (done) {
     return result
   }
 
-  let savedSearches = []
-  if (data) {
-    savedSearches = data.savedSearches
-    if (_isEmpty(savedSearches)) {
-      if (selectedSearch) {
-        // Clear selection
-        setSelectedSearch(null)
-      }
-    } else if (!savedSearches.includes(selectedSearch)) {
-      // Select first one
-      setSelectedSearch(savedSearches[0])
-    }
-  }
-
   return (
     <Fieldset title="Saved searches">
       <Messages error={stateError} />
-      <Form.Group as={Row} className="mb-3" controlId="savedSearchSelect">
-        <Form.Label column sm={2}>
-          <b>Select a saved search</b>
-        </Form.Label>
-        <Col sm={10}>
-          <Form.Select onChange={onSaveSearchSelect}>
-            {savedSearches &&
-              savedSearches.map(savedSearch => (
-                <option value={savedSearch.uuid} key={savedSearch.uuid}>
-                  {savedSearch.name}
-                </option>
+      <UltimatePaginationTopDown
+        componentClassName="searchPagination"
+        className="float-end"
+        pageNum={pageNum}
+        pageSize={DEFAULT_PAGESIZE}
+        totalCount={totalCount}
+        goToPage={setPageNum}
+      >
+        {paginatedSearches.length > 0 ? (
```

Comment:
Suggest to reverse this condition, as the 'else' (which is only one line) is now dangling way below.